### ISSUE TITLE
Black screen in locked session on Manjaro XFCE 20.2.1

### ISSUE THREAD

**Issue by scrawnynewt** [Ylianst/MeshCentral#2312]
Hello!

First of all, I would like to thank you for creating this amazing tool and for all of your efforts.

# The problem

Everything runs well and I was able to add and connect to all of my machines.

Connecting to machines running Ubuntu 20.04.2 LTS works flawlessly.

On machines that run Manjaro XFCE 20.2.1, I can connect to the desktop when no user is logged-in or when a user is already logged-in.
However, when a machine is locked I can still connect to the desktop but all I can see is a black screen and the machine does not react to any input.
The server's logs do not indicate any errors.

I tried to enable the user consent options, i.e., "Notify user", "Prompt for user consent", "Show connection toolbar", (individually and in any possible combination) to provoke the interface to refresh, but it did not resolve the issue.
On the contrary, when I enable the option "Prompt for user consent" the access is instantly denied without prompting the user to accept/deny showing the following message:

> Waiting for user to grant access...
> Denied

Notice that the latter happens for terminal and files as well, regardless if a user is logged-in.

Thank you in advance and let me now if you need any additional information.

# The configuration

Although it might be irrelevant, I am listing my configuration for reference.

## Server

I am running a machine with Ubuntu 20.04.2 LTS with MeshCentral 0.7.72.

## Apache

After a lot of research I managed to successfully setup MeshCentral with reverse proxy behind my Apache server according to the instructions [here](https://github.com/Ylianst/MeshCentral/issues/317) with a Let's Encrypt certificate.
Below you can see my configuration:

```
<VirtualHost *:80>
  ServerName mesh.domain.com

  RewriteEngine on
  RewriteCond %{SERVER_NAME} =mesh.domain.com
  RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

<VirtualHost *:443>
  ServerName mesh.domain.com
  ServerAdmin user@domain.com

  RewriteEngine on
  RewriteCond %{HTTP:Upgrade} websocket [NC]
  RewriteCond %{HTTP:Connection} upgrade [NC]
  RewriteRule . "ws://127.0.0.1:446%{REQUEST_URI}" [P]

  ProxyPass / http://127.0.0.1:446/
  ProxyPassReverse / http://127.0.0.1:446/
  ProxyPreserveHost on
  ProxyAddHeaders off

  ErrorLog ${APACHE_LOG_DIR}/error.log
  CustomLog ${APACHE_LOG_DIR}/access.log combined

  SSLEngine on
  SSLProxyEngine on
  SSLProtocol TLSv1.2
  SSLCipherSuite EECDH+AESGCM:EDH+AESGCM
  SSLHonorCipherOrder on

  <Directory /usr/lib/cgi-bin>
    SSLOptions +StdEnvVars
    Options FollowSymLinks
    AllowOverride All
  </Directory>

  SSLCertificateFile /etc/letsencrypt/live/mesh.domain.com/fullchain.pem
  SSLCertificateKeyFile /etc/letsencrypt/live/mesh.domain.com/privkey.pem
  Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
```

## Docker

Here is my docker-compose file according to the instructions [here](https://github.com/uldiseihenbergs/meshcentral-builds/issues/1) and [here](https://github.com/uldiseihenbergs/meshcentral-builds/issues/2):

```
version: '2'
services:
  meshcentral:
    container_name: meshcentral
    restart: always
    image: uldiseihenbergs/meshcentral:latest
    hostname: mesh.domain.com
    ports:
      - 86:80
      - 446:443
      - 455:8443
    volumes:
      - /srv/meshcentral/meshcentral-files:/home/node/meshcentral/meshcentral-files
      - /srv/meshcentral/meshcentral-data:/home/node/meshcentral/meshcentral-data
```

## MeshCentral

Finally, this is my MeshCentral config.json file:

```
{
  "settings": {
    "cert": "mesh.domain.com",
    "wanonly": true,
    "sessionKey": "randomString",
    "_port": 443,
    "_aliasPort": 443,
    "tlsoffload": "127.0.0.1",
    "selfupdate": false,
    "minify": true,
    "exactports": true
  },
  "domains": {
    "": {
      "title": "MeshCentral",
      "newaccounts": false,
      "certurl": "my.server.static.ip"
    }
  },
  "smtp": {
    "host": "mail.domain.com",
    "port": 587,
    "from": "mesh@domain.com",
    "user": "mesh@domain.com",
    "pass": "password",
    "tls": false
  }
}
```

**Comment by krayon007** [comment]
For your user consent issue. Are you saying it happens when a user is logged in with a locked desktop, or are you saying user consent just never works on your system?

**Comment by scrawnynewt** [comment]
The option "Prompt for user consent" never works for any user consent setting category, i.e., "Desktop", "Terminal", "Files", on Manjaro XFCE 20.2.1.

**Comment by krayon007** [comment]
> The option "Prompt for user consent" never works for any user consent setting category, i.e., "Desktop", "Terminal", "Files", on Manjaro XFCE 20.2.1.

Interesting. I have Manjaro 18 with XFCE, and I can't reproduce your issue. By any chance do you not have Zenity installed on your Manjaro system? Just for giggles, try downloading the binary installer, and run that, and let me know if that gives you an error message, or if it displays a dialog box.

**Comment by scrawnynewt** [comment]
> By any chance do you not have Zenity installed on your Manjaro system?

No, I didn't.
I can verify that installing Zenity prompts the user to accept/deny the connection and solves the corresponding issue.

> Just for giggles, try downloading the agent binary installer, and run that, and let me know if that gives you an error message, or if it displays a dialog box.

I tried this, it displays a dialog box now.

**Comment by scrawnynewt** [comment]
Today I set up some VMs to try the agent on a fresh Manjaro XFCE 20.2.1 installation, in order to make sure that the issue is not due to my machines' configuration.
Eventually, I came across the exact same issue, i.e., when the session is locked the desktop does not respond to mouse/keyboard input from MeshCentral.
My guess is that the issue is probably related to LightDM 1.30 or LightDM GTK+ Greeter 2.0.8.

For reference, I did the same on fresh Manjaro GNOME and KDE 20.2.1 installations and did not come across any issues.
